### PR TITLE
Simplify code for dispatching events

### PR DESCRIPTION
**Pull request by robert-ancell**

### CODE REVIEW

**Comment by stuartmorgan** [comment]
test-exempt: code refactor with no semantic change

File: shell/platform/linux/fl_keyboard_handler.cc [hunk 1]
```diff
@@ -538,14 +520,16 @@ gboolean fl_keyboard_handler_handle_event(FlKeyboardHandler* self,
...
-      .specified_logical_key = fl_keyboard_layout_get_logical_key(
-          self->derived_layout, fl_key_event_get_group(event),
-          fl_key_event_get_keycode(event)),
-      .user_data = user_data,
-  };
-  g_ptr_array_foreach(self->responder_list, dispatch_to_responder, &data);
+  uint64_t specified_logical_key = fl_keyboard_layout_get_logical_key(
+      self->derived_layout, fl_key_event_get_group(event),
+      fl_key_event_get_keycode(event));
+  for (guint i = 0; i < self->responder_list->len; i++) {
```
Comment (cbracken) [inline shell/platform/linux/fl_keyboard_handler.cc:526]: Presumably `responder_list` will never be null here?

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

File: shell/platform/linux/fl_keyboard_handler.cc (hunk 1 above)
Comment (robert-ancell) [reply shell/platform/linux/fl_keyboard_handler.cc:526]: Correct, it is created in the constructor.

**Pull request merged**